torchgeo/datasets/utils.py

```diff
@@ -722,3 +724,33 @@ def which(name: Path) -> Executable:
     else:
         msg = f'{name} is not installed and is required to use this dataset.'
         raise DependencyNotFoundError(msg) from None
+
+
+def convert_poly_coords(
+    geom: shapely.geometry.shape, affine_obj: affine.Affine, inverse: bool = False
+) -> shapely.geometry.shape:
+    """Convert geocoordinates to pixel coordinates and vice versa, based on `affine_obj`.
+
+    Args:
+        geom: shapely.geometry.shape to convert
+        affine_obj: affine.Affine object to use for geoconversion
+        inverse: If true, convert geocoordinates to pixel coordinates
+
+    Returns:
+        shapely.geometry.shape: input shape converted to pixel coordinates
```

Comment:
No need to include types in docstrings, Sphinx automatically extracts those from the type hints.

Also add:
```rst
.. versionadded:: 0.8
```